cvat/apps/dataset_manager/task.py

```diff
@@ -388,8 +389,12 @@ def _save_tags_to_db(self, tags):
         self.ir_data.tags = tags
 
     def _set_updated_date(self):
-        self.db_job.segment.task.touch()
-        self.db_job.touch()
+        db_task = self.db_job.segment.task
+        with transaction.atomic():
+            if db_project := db_task.project:
+                db_project.touch()
+            db_task.touch()
+            self.db_job.touch()
```

Comment:
I'd suggest to reverse the touch order here, since it ought to be that `project.updated_date >= task.updated_date >= job.updated_date`.